### DISCUSSION TITLE
Managing translations for dynamic string

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#16412] [Q&A] [answered]
I have a question regarding OrchardCore's localization capabilities.
Is it possible to manage translations for dynamic strings, such as HtmlPart?
If so, could you please provide some guidance or resources on how to achieve this?

**Comment by hishamco** [comment, marked as answer]
The dynamic localization is not supported yet #4466, but you could try https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/main/src/OrchardCoreContrib.DataLocalization/README.md

**Reply by DotCat1985** [reply]
As you can see on the screenshot below, I set the content fields resources of my own content type.

<img width="488" alt="Screenshot 2024-07-10 at 17 09 18" src="https://github.com/OrchardCMS/OrchardCore/assets/61869142/ddd8abab-ad63-4c65-a3a7-16c5704e5ae9">

Could I get them by using the following codes?

`{{ "City" | t }}`
`{{ "Country" | t }}`

**Reply by hishamco** [reply]
What are you trying to do? Coz the above liquid is not valid yet for the mentioned module

**Reply by DotCat1985** [reply]
How can I obtain the values on the screenshot above?

**Reply by hishamco** [reply]
Do you mean CItta, Nazione .. etc?

**Reply by DotCat1985** [reply]
> Do you mean CItta, Nazione .. etc?

The italian translation of the resources names "City", "Country", "E-mail".

**Reply by hishamco** [reply]
Please check the end of my talk https://www.youtube.com/watch?v=WQbpRHV9ugw&t=16s&pp=ygUUb3JjaGFyZCBjb3JlIGhhcnZlc3Q%3D

**Reply by hishamco** [reply]
1. Inject `IDataLocalizer`
2. Access the localized content using `D["Country", "ContentType"]`
